File: rest-service/src/main/java/mb/lib/db/model/DBJobStatus.java
package mb.lib.db.model;

@Deprecated
public enum DBJobStatus
{
  Completed("completed"),
  Running("running"),
  Errored("errored"),
  Queued("queued");

  public final String value;

  DBJobStatus(String value) {
    this.value = value;
  }

  @Override
  public String toString() {
    return name();
  }

  public static DBJobStatus fromString(String value) {
    for (var v : values())
      if (v.value.equals(value))
        return v;

    throw new IllegalArgumentException(String.format("Invalid JobStatus value \"%s\"", value));
  }
}

File: rest-service/src/main/kotlin/mb/lib/db/model/DBJobStatus.kt
package mb.lib.db.model

@Deprecated("")
enum class DBJobStatus(val value: String) {
  Completed("completed"), Running("running"), Errored("errored"), Queued("queued");

  override fun toString(): String {
    return name
  }

  companion object {
    fun fromString(value: String): DBJobStatus {
      for (v in values()) if (v.value == value) return v
      throw IllegalArgumentException(String.format("Invalid JobStatus value \"%s\"", value))
    }
  }
}